tests/sentry/integrations/github_enterprise/test_client.py

```diff
@@ -0,0 +1,154 @@
+                "icon": "https://github.example.org/avatar.png",
+                "domain_name": "github.example.org/Test-Organization",
+                "account_type": "Organization",
+                "installation_id": "install_id_1",
+                "installation": {
+                    "client_id": "client_id",
+                    "client_secret": "client_secret",
+                    "id": "2",
+                    "name": "test-app",
+                    "private_key": "private_key",
+                    "url": "github.example.org",
+                    "webhook_secret": "webhook_secret",
+                    "verify_ssl": True,
+                },
+            },
+        )
+        self.repo = Repository.objects.create(
+            organization_id=self.organization.id,
+            name="Test-Organization/foo",
+            url="https://github.example.org/Test-Organization/foo",
+            provider="integrations:github_enterprise",
+            external_id=123,
+            integration_id=integration.id,
+        )
+        self.install = integration.get_installation(organization_id=self.organization.id)
+        self.client = self.install.get_client()
+        responses.add(
+            method=responses.POST,
+            url=f"{self.base_url}/app/installations/install_id_1/access_tokens",
+            body='{"token": "12345token", "expires_at": "2030-01-01T00:00:00Z"}',
+            status=200,
+            content_type="application/json",
+        )
+
+    @responses.activate
+    def test_check_file(self):
+        path = "src/sentry/integrations/github/client.py"
+        version = "master"
+        url = f"{self.base_url}/repos/{self.repo.name}/contents/{path}?ref={version}"
+
+        responses.add(
+            method=responses.HEAD,
+            url=url,
+            json={"text": 200},
+        )
+
+        assert hasattr(self.client, "check_file")
+        resp = self.client.check_file(self.repo, path, version)
+        assert resp.status_code == 200
+
+    @responses.activate
+    def test_check_no_file(self):
+        path = "src/santry/integrations/github/client.py"
+        version = "master"
+        url = f"{self.base_url}/repos/{self.repo.name}/contents/{path}?ref={version}"
+
+        responses.add(method=responses.HEAD, url=url, status=404)
+
+        with pytest.raises(ApiError):
+            assert hasattr(self.client, "check_file")
+            self.client.check_file(self.repo, path, version)
+        assert responses.calls[1].response.status_code == 404
+
+    @responses.activate
+    def test_get_stacktrace_link(self):
+        path = "/src/sentry/integrations/github/client.py"
+        version = "master"
+        url = f"{self.base_url}/repos/{self.repo.name}/contents/{path.lstrip('/')}?ref={version}"
+
+        responses.add(
+            method=responses.HEAD,
+            url=url,
+            json={"text": 200},
+        )
+
+        assert isinstance(self.install, RepositoryMixin)
```

Comment:
it's a little weird to assert the type of a mixin -- this should assert the type of a concrete class